src/DynamoRevit/DynamoRevit.cs

```diff
@@ -890,4 +912,95 @@ public override IEnumerable<string> GetDynamoUserDataLocations()
             return paths;
         }
     }
+
+    /// <summary>
+    /// Defines parameters used for loading internal Dynamo Revit packages
+    /// </summary>
+    [Serializable()]
+    public class InternalPackage
+    {
+        /// <summary>
+        /// keeps the path to the node file
+        /// </summary>
+        public string NodePath { get; set; }
+
+        /// <summary>
+        /// keeps the path to the layoutSpecs.json file
+        /// </summary>
+        public string LayoutSpecsPath { get; set; }
+    }
+    internal static class DynamoRevitInternalNodes
+    {
+        private static IEnumerable<string> GetAllInternalPackageFiles()
+        {
+            string currentAssemblyPath = Assembly.GetExecutingAssembly().Location;
+            string currentAssemblyDir = Path.GetDirectoryName(currentAssemblyPath);
+
+            string internalNodesDir = Path.Combine(currentAssemblyDir, "nodes");
```

Comment:
Would you split `nodes` out to be a const string?

Reply:
done